invalid init for array cause compiler fail

### Describe the bug

see code

### Reproduction Steps

mix.v
```v
z := [2][]?int{init: []?int{len: 3}}
dump(z)
```


### Expected Behavior

compile ok or report error msg

### Current Behavior

```
v mix.v
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/mix.01KERP4CW1AWD22TQ1XVSY2M77.tmp.c:5504: error: 'te' undeclared
=================================================================
Try passing `-g` when compiling, to see a .v file:line information, that correlates more with the C error.
(Alternatively, pass `-show-c-output`, to print the full C error message).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.5.0 c98321ad238f9a18989d1e9d32d71bedd3ee95b7.e6e28f1

### Environment details (OS name and version, etc.)

|V full version      |V 0.5.0 c98321ad238f9a18989d1e9d32d71bedd3ee95b7.e6e28f1
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 24.04.3 LTS
|Processor           |8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-9700 CPU @ 3.00GHz
|Memory              |9.77GB/15.51GB
|                    |
|V executable        |/media/HD/github/kbkpbot/v/v
|V last modified time|2026-01-12 08:45:34
|                    |
|V home dir          |OK, value: /media/HD/github/kbkpbot/v
|VMODULES            |OK, value: /home/mars/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/mars/v/bug/arr
|                    |
|Git version         |git version 2.43.0
|V git status        |weekly.2025.52-154-ge6e28f16-dirty
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|gcc version         |gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|clang version       |Ubuntu clang version 18.1.3 (1ubuntu1)
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.39-0ubuntu8.6) 2.39


> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -198,25 +198,11 @@ fn (mut g Gen) fixed_array_init(node ast.ArrayInit, array_type Type, var_name st
 		}
 	} else if node.has_init {
 		info := array_type.unaliased_sym.info as ast.ArrayFixed
-		if node.elem_type.has_flag(.option) {
-			for i in 0 .. info.size {
-				g.expr_with_init(node)
-				g.add_commas_and_prevent_long_lines(i, info.size)
-			}
-		} else {
-			if g.table.final_sym(info.elem_type).kind in [.struct, .interface] {
-				for i in 0 .. info.size {
-					g.expr_with_init(node)
-					g.add_commas_and_prevent_long_lines(i, info.size)
-				}
-			} else {
-				before_expr_pos := g.out.len
-				{
-					g.expr_with_init(node)
-				}
-				sexpr := g.out.cut_to(before_expr_pos)
-				g.write_c99_elements_for_array(info.size, sexpr)
-			}
+		// Always loop and call expr_with_init for each element to avoid copying
+		// temporary variable declarations and initialization code
+		for i in 0 .. info.size {
+			g.expr_with_init(node)
+			g.add_commas_and_prevent_long_lines(i, info.size)
 		}
 	} else if is_amp {
 		g.write('0')

--- a/vlib/v/tests/fixed_array_init_with_option_array_test.v
+++ b/vlib/v/tests/fixed_array_init_with_option_array_test.v
@@ -1,0 +1,6 @@
+module main
+
+fn test_fixed_array_init_with_option_array() {
+	z := [2][]?int{init: []?int{len: 3}}
+	assert z.str() == '[[Option(none), Option(none), Option(none)], [Option(none), Option(none), Option(none)]]'
+}

